cluster.yml

```diff
@@ -30,15 +30,24 @@
     - { role: docker, tags: docker }
     - { role: rkt, tags: rkt, when: "'rkt' in [ etcd_deployment_type, kubelet_deployment_type ]" }
 
-- hosts: etcd:!k8s-cluster
+- hosts: all
+  any_errors_fatal: true
+  roles:
+    - { role: vault, tags: vault, vault_bootstrap: true, when: "cert_management == 'vault'" }
+
+- hosts: etcd:k8s-cluster
   any_errors_fatal: true
   roles:
     - { role: etcd, tags: etcd }
 
+- hosts: etcd:k8s-cluster:vault
+  any_errors_fatal: true
+  roles:
+    - { role: vault, tags: vault, when: "cert_management == 'vault'"}
+
 - hosts: k8s-cluster
   any_errors_fatal: true
   roles:
-    - { role: etcd, tags: etcd }
```

Comment:
that doesn't work that way, please don't change that. This step must deploy etcd nodes co-located within the k8s cluster.